salesforce/bulk.go

```diff
@@ -109,7 +110,7 @@ func (c *Connector) BulkWrite( //nolint:funlen,cyclop
 	config BulkOperationParams,
 ) (*BulkOperationResult, error) {
 	// Only support upsert for now
-	switch config.Mode {
+	switch config.Mode { //nolint:exhaustive
```

Comment:
Instead of a switch statement, this would be better written as an if statement:
```
if config.Mode != Upsert {
 return nil, fmt.Errorf("%w: %s", ErrUnsupportedMode, config.Mode)
}
```